Lib/http/server.py

```diff
@@ -967,25 +976,33 @@ def _get_best_family(*address):
     return family, sockaddr
 
 
+def _make_server(HandlerClass=BaseHTTPRequestHandler,
+                 ServerClass=ThreadingHTTPServer,
+                 protocol="HTTP/1.0", port=8000, bind=None,
+                 tls_cert=None, tls_key=None, tls_password=None):
+    ServerClass.address_family, addr = _get_best_family(bind, port)
+    HandlerClass.protocol_version = protocol
+
+    if tls_cert:
+        return ServerClass(addr, HandlerClass, certfile=tls_cert,
+                           keyfile=tls_key, password=tls_password)
+    else:
+        return ServerClass(addr, HandlerClass)
+
+
 def test(HandlerClass=BaseHTTPRequestHandler,
          ServerClass=ThreadingHTTPServer,
          protocol="HTTP/1.0", port=8000, bind=None,
          tls_cert=None, tls_key=None, tls_password=None):
     """Test the HTTP request handler class.
 
     This runs an HTTP server on port 8000 (or the port argument).
-
     """
-    ServerClass.address_family, addr = _get_best_family(bind, port)
-    HandlerClass.protocol_version = protocol
-
-    if tls_cert:
-        server = ServerClass(addr, HandlerClass, certfile=tls_cert,
-                             keyfile=tls_key, password=tls_password)
-    else:
-        server = ServerClass(addr, HandlerClass)
-
-    with server as httpd:
+    with _make_server(
+        HandlerClass=HandlerClass, ServerClass=ServerClass,
+        protocol=protocol, port=port, bind=bind, tls_cert=tls_cert,
+        tls_key=tls_key, tls_password=tls_password
```

Comment:
```suggestion
        HandlerClass=HandlerClass, ServerClass=ServerClass,
        protocol=protocol, port=port, bind=bind,
        tls_cert=tls_cert, tls_key=tls_key, tls_password=tls_password,
```

I would prefer this split just to group the tls_* arguments together. I *think* it fits on 80 chars.

Reply:
Fixed in 64122df4e181601122cc0afcdfff6163d9aa690d. It ends up being only 68 chars so it fits easily under 80. I didn't leave in a trailing comma, if that matters I can add it in.